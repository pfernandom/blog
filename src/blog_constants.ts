import path from 'path'

/**
 * src/content
 */
export const CONTENT_PATH = path.join('src', 'app', 'content')

/**
 * content
 */
export const CONTENT_RELATIVE_PATH = path.join('app', 'content')
