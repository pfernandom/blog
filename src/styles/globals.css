html,
body {
  padding: 0;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen,
    Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
  /* font: 100%/1.75 "Avenir Next", "Helvetica Neue", "Segoe UI", "Helvetica",
    "Arial", sans-serif; */
}

footer {
  text-align: center;
}

code {
  background: #f5f2f0;
  padding: 0em 0.5em;
  display: inline-block;
  font-family: Roboto;
  font-size: 1.1em;
}

@media (prefers-color-scheme: dark) {
  html {
    color-scheme: dark;
  }
  body {
    color: white;
    background: black;
  }
  code {
    background-color: grey;
  }
  pre {
    filter: brightness(0.95);
  }
}

h1,
h2,
h3,
h4 {
  font-family: Montserrat, sans-serif;
  font-weight: 500;
  text-rendering: optimizeLegibility;
  line-height: 1.1;
}

h2,
h3,
h4 {
  margin-top: 2em;
}

h1 {
  font-size: 2em;
  color: var(---heading-color);
}

h2 {
  font-size: 1.5em;
}

h2,
h3 {
  color: var(---heading-color);
  filter: brightness(0.8);
}

a:hover {
  text-decoration: underline;
}

p {
  margin-bottom: 1.5em;
}

.layout-content {
  max-width: 900px;
  margin: 1em auto;
  padding: 0 2em;
}

a {
  color: var(---heading-color);
}

a:hover {
  transform: scale(1.05);
  display: inline-block;
}

@import url(./blog.css);
@import url(./instagram.css);
