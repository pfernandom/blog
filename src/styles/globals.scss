html,
body {
  padding: 0;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen,
    Ubuntu, Cantarell, 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif;

  /* font: 100%/1.75 "Avenir Next", "Helvetica Neue", "Segoe UI", "Helvetica",
    "Arial", sans-serif; */
}

footer {
  text-align: center;
}
/* 
code {
  background: #f5f2f0;
  padding: 0 0.5em;
  display: inline-block;
  font-family: Roboto, sans-serif;
  font-size: 1.1em;
} */

:not(pre) code {
  margin: 1px;
  padding-left: 1px;
  padding-right: 1px;
}

@media (prefers-color-scheme: dark) {
  html {
    color-scheme: dark;
  }

  body {
    color: white;
    background: black;
  }

  code {
    background-color: grey;
  }

  pre {
    filter: brightness(0.95);
  }
}

h1,
h2,
h3,
h4 {
  font-family: Montserrat, sans-serif;
  font-weight: 500;
  text-rendering: optimizelegibility;
  line-height: 1.1;
}

h2,
h3,
h4 {
  margin-top: 2em;
}

h1 {
  font-size: 2em;
  color: var(---heading-color);
}

h2 {
  font-size: 1.5em;
}

h2,
h3,
h4 {
  color: var(---heading-color);
  filter: brightness(0.8);
}

a:hover {
  text-decoration: underline;
}

p {
  margin-bottom: 1.5em;
}

.layout-content {
  max-width: 900px;
  margin: 1em auto;
  padding: 0 2em;
}

a {
  color: var(---heading-color);
  text-decoration: none;
}

a:hover {
  text-decoration: underline;
}

@import './blog.scss';
@import './instagram.scss';

blockquote {
  --background: #fff;
  --color: #00cc8f;
  --font-size: 1.2em;

  /* padding: 0.5em;
  background-color: gainsboro;
  font-family: Courier; */
  border-left: 3px solid var(--color);
  color: #1a1a1a;
  font-family: Georgia, Times, 'Times New Roman', serif;
  font-size: var(--font-size);
  font-style: italic;
  line-height: 1.8em;
  margin-block-start: 0;
  margin-block-end: 0;
  margin-left: 0;
  padding: 1em 2em;
  position: relative;
  transition: 0.2s border ease-in-out;
  z-index: 0;
  width: 80%;
}

blockquote p {
  margin: 0;
  margin-bottom: 0.5em;
}

blockquote::before {
  content: '';
  position: absolute;
  top: 50%;
  left: -4px;
  height: 2em;
  background-color: var(--background);
  width: 5px;
  margin-top: -1em;
}

blockquote::after {
  content: '\201C';
  position: absolute;
  top: 50%;
  left: -0.5em;
  color: var(--color);
  font-style: normal;
  font-size: 2em;
  font-weight: bold;
  line-height: 1.5em;
  text-align: center;
  text-indent: -2px;
  width: 1em;
  margin-top: -0.5em;
  transition: 0.2s all ease-in-out, 0.4s transform ease-in-out;
}

blockquote.active:after {
  transform: rotateY(360deg);
}

blockquote h3 {
  margin-top: 0;
  font-size: var(--font-size);
}
