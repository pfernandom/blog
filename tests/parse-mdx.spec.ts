import { describe, expect, test } from '@jest/globals'
import fs from 'fs'
import parseMDX from 'app/content-manager/helpers/parse_mdx'
import { CONTENT_PATH } from 'blog_constants'

describe('parseMDX', function () {
  test('it works correctly', async function () {
    const imagesPath = 'public/opt_images/content/software/1988/0_test_post'
    const testPost = CONTENT_PATH + '/software/1988/0_test_post/index.mdx'
    const fileContents = fs.readFileSync(testPost, 'utf8')

    const { postData, content } = parseMDX({
      fileContents,
      imagesPath,
    })

    expect(postData.test).toBe(true)
    expect(fs.existsSync(postData.hero_image)).toBe(true)
    expect(postData).toMatchSnapshot()
    expect(content).toMatchSnapshot()
  })
})
