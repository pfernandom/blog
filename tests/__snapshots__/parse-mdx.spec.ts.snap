// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseMDX it works correctly 1`] = `
{
  "date": "June 19, 1988",
  "description": [
    "line 1",
    "line 2",
  ],
  "hero_image": "public/opt_images/content/software/1988/0_test_post/hero.webp",
  "hero_image_alt": "A hero image for this post about state management",
  "hero_image_blur": "public/opt_images/content/software/1988/0_test_post/blur_hero.webp",
  "hero_image_original": "public/opt_images/content/software/1988/0_test_post/hero.jpeg",
  "key_words": [
    "test",
    "post",
  ],
  "legacy": true,
  "published": true,
  "series": null,
  "social_footer": "Visit pedromarquez.dev for the full post",
  "social_subtitle": "line 1",
  "social_title": "This is a test post",
  "test": true,
  "title": "This is a test post",
}
`;

exports[`parseMDX it works correctly 2`] = `
"
import heroImage from './demo.webp'

# Test post

This is only a test

<Image src={heroImage} alt="Gif showing the server running" />
"
`;
